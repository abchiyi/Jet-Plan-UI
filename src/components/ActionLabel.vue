<template>
  <ActionFeedback tag="label" active class="demo action-label">
    <slot></slot>
  </ActionFeedback>
</template>
<script lang="ts" setup>
import { ActionFeedback } from "jet-plan-ui";
</script>

<style>
.action-label {
  justify-content: space-between;
  align-items: center;
  display: flex;
}

.action-label > * {
  margin-bottom: unset !important;
  margin-top: unset !important;
}

.action-label * {
  user-select: none;
}
</style>
