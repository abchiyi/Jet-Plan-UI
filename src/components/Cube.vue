<template>
  <m-row :class="classes" center middle>
    <slot></slot>
  </m-row>
</template>

<script>
export default {
  name: "m-cube",
  props: {
    dark: {
      type: Boolean,
      default: false,
    },
  },
  computed: {
    classes() {
      let classes = ["m-cube"];
      if (this.dark) classes.push("dark");
      return classes;
    },
  },
};
</script>

<style>
.m-cube {
  width: 100px !important;
  height: 100px;
  border-radius: 4px;
  background-color: #1687a7;
  /* 文本设置 */
  font-size: 20px;
  color: #fff;
  transition: 0.4s ease-in-out;
}
.m-cube.dark {
  background-color: #0b81a1;
}
</style>
