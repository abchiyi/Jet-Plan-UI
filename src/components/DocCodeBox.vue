<script lang="ts">
import { defineComponent, h, isVNode } from "vue";
import { CodeBox } from "jet-plan-ui";
export default defineComponent({
  methods: {
    getLanguage() {
      const pre = this.$slots.default?.()[0];
      if (pre?.children instanceof Array) {
        const code = pre.children[0];

        if (isVNode(code)) {
          const codeClass = code.props?.class as string;
          return {
            language: codeClass.replace("language-", ""),
            autodetect: false,
          };
        }
      }

      return {
        language: "",
        autodetect: true,
      };
    },
  },
  render() {
    return h(
      CodeBox,
      {
        canCopy: true,
        ...this.getLanguage(),
      },
      {
        default: () => this.$slots.default?.(),
        copyButton: () => h("i", { class: "bi bi-clipboard-plus" }),
      }
    );
  },
});
</script>
