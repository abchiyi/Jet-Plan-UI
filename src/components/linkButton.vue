<script lang="ts">
import { defineComponent, h } from "vue";
import { Button } from "jet-plan-ui";
import { RouterLink, useLink } from "vue-router";
export default defineComponent({
  name: "link-button",
  props: {
    //@ts-ignore
    ...RouterLink.props,
  },
  setup(props) {
    return {
      //@ts-ignore
      link: useLink(props),
    };
  },
  render() {
    return h(
      Button,
      {
        onClick: this.link.navigate,
        class: "link-button",
        href: this.link.href.value,
        tag: "a",
      },
      {
        default: this.$slots.default,
      }
    );
  },
});
</script>

<style>
.link-button {
  text-decoration-line: none;
}
</style>
