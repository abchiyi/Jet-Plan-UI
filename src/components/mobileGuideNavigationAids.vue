<template>
  <!-- sub app bar -->
  <j-row
    class="app-bar frosted-glass"
    spaceMode="between"
    id="sub-app-bar"
    v-show="Mobile"
    Y="center"
  >
    <j-button class="button-menu" @click="clickMenu" text>
      <i class="bi bi-list"></i>
      菜单
    </j-button>
  </j-row>
  <!-- drawer left -->
  <j-drawer id="drawer-left" from="left" :expand="drawerLeftExpand">
    <slot name="drawer"></slot>
  </j-drawer>
  <!-- drawer mask -->
  <div
    v-show="drawerLeftExpand"
    @click="drawerLeftExpand = false"
    id="drawer-mask"
  ></div>
</template>
<script lang="ts">
import { Row as JRow, Button as JButton, Drawer as JDrawer } from "jet-plan-ui";
import { shadowPainter } from "jet-plan-ui/tool";
import { inject, defineComponent } from "vue";
export default defineComponent({
  components: {
    JRow,
    JButton,
    JDrawer,
  },
  setup() {
    return {
      Mobile: inject<string>("Mobile"),
    };
  },
  methods: {
    clickMenu() {
      this.drawerLeftExpand = !this.drawerLeftExpand;
    },
  },
  data() {
    return {
      drawerShadow: shadowPainter("right", 2, "#76909233"),
      drawerLeftExpand: false,
    };
  },
});
</script>
<style scoped>
/* Sub app bar */
#sub-app-bar {
  height: var(--HEIGHT-HEADER);
  box-sizing: border-box;
  position: sticky;
  width: 100vw;
  top: -1px;
  margin-left: -1rem;
}

#drawer-left {
  box-shadow: v-bind(drawerShadow);
  background: var(--background);
  width: 255px;
}

#drawer-mask {
  backdrop-filter: blur(2px);
  position: fixed;
  height: 100vh;
  width: 100vw;
  left: 0;
  top: 0;
}
</style>
