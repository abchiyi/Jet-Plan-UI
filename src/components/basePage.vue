<template>
  <div class="base-page" :class="classes">
    <header id="app-bar" class="app-bar frosted-glass">
      <j-row Y="center" id="header-group" space-mode="between">
        <logo-title id="logo-title" />

        <div id="nav-group">
          <!-- 导航菜单按钮 -->
          <j-button
            @click="toOpenDropMenu"
            title="open-drop-menu"
            id="open-drop-menu"
            v-show="Mobile"
            mode="text"
          >
            <i :class="['bi', drawerTopOpen ? 'bi-x' : 'bi-list']"></i>
          </j-button>

          <!-- 导航链接 -->
          <div id="nav" v-show="!Mobile">
            <!-- TODO dark mode switch -->
            <link-button to="/about" aria-label="关于" mode="text">
              关于
            </link-button>

            <link-button to="/component-list" aria-label="组件列表" mode="text">
              组件列表
            </link-button>

            <toGiteeIcoButton />
            <toGithubIcoButton />
          </div>
        </div>
      </j-row>
    </header>

    <j-row id="content-shell" X="center">
      <div id="sidebar-left" v-show="sidebarLeftShow">
        <slot name="sidebarLeft"></slot>
      </div>
      <div id="content">
        <main>
          <slot></slot>
        </main>
        <j-row tag="footer" Y="center" X="center">
          <p class="license">
            Released under the Apache 2.0 License. Created by abchiyi
          </p>
        </j-row>
      </div>
      <div id="sidebar-right" v-show="Desktop && $slots.sidebarRight">
        <slot name="sidebarRight"></slot>
      </div>
    </j-row>

    <j-drawer :expand="drawerTopOpen" id="drawer-top" from="top" opacity>
      <ul id="nav-link-group-drop-menu">
        <li>
          <link-button to="/about" aria-label="关于">关于</link-button>
        </li>
        <li>
          <link-button to="/component-list" aria-label="组件列表">
            组件列表
          </link-button>
        </li>
      </ul>
      <hr />
      <div id="social-links">
        <toGiteeIcoButton />
        <toGithubIcoButton />
      </div>
    </j-drawer>
  </div>
</template>

<script lang="ts">
import { Row as JRow, Button as JButton, Drawer as JDrawer } from "jet-plan-ui";
import { getWindowWidthIn, shadowPainter } from "jet-plan-ui/tool";

import toGiteeIcoButton from "./toGiteeIcoButton.vue";
import toGithubIcoButton from "./toGithubIcoButton.vue";
import linkButton from "./linkButton.vue";
import logoTitle from "./logoTitle.vue";
import { defineComponent, computed, provide } from "vue";

export default defineComponent({
  name: "base-page",
  setup() {
    const wbp = getWindowWidthIn();

    const Mobile = computed(() => {
      return ["xs", "sm"].includes(wbp.value);
    });
    provide("Mobile", Mobile);

    const Tablet = computed(() => {
      return ["md", "lg"].includes(wbp.value);
    });
    provide("Tablet", Tablet);

    const Desktop = computed(() => {
      return ["xl"].includes(wbp.value);
    });

    provide("Desktop", Desktop);

    return {
      wbp,
      Mobile,
    };
  },
  components: {
    linkButton,
    JButton,
    JDrawer,
    JRow,
    logoTitle,
    toGiteeIcoButton,
    toGithubIcoButton,
  },
  data() {
    return {
      // TODO 使用主题定义色彩
      headerShadow: shadowPainter("bottom", 2, "#76909233"),
      DrawerTopOpen_data: false,
    };
  },
  computed: {
    sidebarLeftShow() {
      if (this.$slots.sidebarLeft && (this.Tablet || this.Desktop)) {
        return true;
      }
      return false;
    },

    drawerTopOpen() {
      return this.Mobile && this.DrawerTopOpen_data;
    },

    Tablet() {
      return ["md", "lg"].includes(this.wbp);
    },
    Desktop() {
      return ["xl"].includes(this.wbp);
    },

    classes() {
      return [
        this.Desktop ? "desktop" : "",
        this.Tablet ? "tablet" : "",
        this.Mobile ? "mobile" : "",
      ];
    },
  },
  watch: {
    Mobile(v) {
      // 屏幕断点 Mobile 为 false 关闭 DrawerLeftOpen_data
      if (!v) this.DrawerTopOpen_data = v;
    },
  },
  methods: {
    toOpenDropMenu() {
      if (this.Mobile) {
        this.DrawerTopOpen_data = !this.DrawerTopOpen_data;
      }
    },
  },
});
</script>
<style>
body {
  overflow-y: scroll;
}
/* 变量设定 */
.base-page {
  --CONTENT-WIDTH-MAIN: 760px;
  --CONTENT-WIDTH: 1400px;
  --WIDTH-DRAWER: 255px;
  --HEIGHT-HEADER: 50px;
  --HEIGHT-FOOTER: 2rem;
  --TOP-SPACE: calc(1rem + var(--HEIGHT-HEADER));
  --HEIGHT-MAIN: calc(100vh - var(--TOP-SPACE) - var(--HEIGHT-FOOTER));
}

/* 侧栏 （左&右）*/
.base-page #sidebar-right,
.base-page #sidebar-left {
  height: var(--HEIGHT-MAIN);
  box-sizing: border-box;
  top: var(--TOP-SPACE);
  position: sticky;
  overflow-y: auto;
  width: 280px;
}

.base-page #sidebar-left {
  border-right: dashed 2px var(--border);
  margin-right: 30px;
}

.base-page #sidebar-right {
  border-left: dashed 2px var(--border);
  margin-left: 30px;
}

/* 下拉抽屉 */
#drawer-top {
  /* 强制应用变量以覆盖组件参数设定 */
  --migration-distance: translate3d(0, -30px, 0) !important;
  padding-top: var(--HEIGHT-HEADER);
  background: var(--background);
  height: 100vh;
}

/* 抽屉内容 */
#nav-link-group-drop-menu {
  padding: unset;
}
#nav-link-group-drop-menu .j-button {
  margin: 0.5rem auto;
  list-style: unset;
  font-size: 1rem;
  display: block;
  width: 255px;
}

/* ----- 文档外壳 */
.base-page > #content-shell {
  /* 页头不会覆盖文档 */
  max-width: var(--CONTENT-WIDTH);
  padding-top: var(--TOP-SPACE);
  margin: 0 auto;
}

/* 取消移动页面页头内边距 */
.base-page.mobile > #content-shell {
  padding-top: 0;
}

/* ----- 文档内容节点(仅正文) */
.base-page #content > main {
  min-height: var(--HEIGHT-MAIN);
  box-sizing: border-box;
  padding: 0 1rem;
}
.base-page #content {
  width: 100%;
}

/* 禁用标题的上外边距,该外边距会超出容器导致页面高度异常 */
.base-page main h1 {
  margin-top: unset;
}

/* ----- 文档内容节点(包含 footer) */
.base-page > #content-shell > #content {
  box-sizing: border-box;
  width: 100%;
}

/* ----- 页头 */
.app-bar {
  box-shadow: v-bind("headerShadow");
  height: var(--HEIGHT-HEADER);
  margin-top: unset !important;
  position: fixed;
  right: 0;
  left: 0;
  top: 0;
}

.base-page.mobile > #app-bar {
  /*使用'sticky'定位于屏幕顶部时避免上边缘暴露下层文档内容。超出屏幕边缘以遮盖*/
  top: calc(-1.5 * var(--HEIGHT-HEADER));
  position: sticky;
}

/* ----- 页头内容组 */
#header-group {
  max-width: var(--CONTENT-WIDTH);
  margin: auto;
  height: 100%;
}

/* ----- 其他杂项 */

/* 使图像对其文本 */
.j-button img,
.j-button svg {
  vertical-align: -0.125em;
}

/* ----- 页头导航链接 */
#nav > * {
  vertical-align: middle;
}

#nav > *:hover {
  color: var(--primary);
}

#nav > .j-button {
  font-size: 1rem;
}

/* ----- 页脚 */
footer {
  height: var(--HEIGHT-FOOTER);
}
footer > .license {
  box-sizing: border-box;
  font-size: 12px;
  height: 1.2em;
  opacity: 0.3;
  margin: 0;
}

/* ----- 下拉抽屉 */

#drop-drawer {
  height: 100vh;
  width: 100vw;
}
/* 社交链接 */
#social-links {
  justify-content: center;
  display: flex;
}
#social-links > * {
  font-size: 24px;
}

/* ----- 开启&关闭 下拉抽屉按钮 */
#open-drop-menu {
  font-size: 1.5rem;
}
</style>
