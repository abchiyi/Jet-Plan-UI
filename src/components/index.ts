import linkButton from "./linkButton.vue";
import previewDemoCode from "./previewDemoCode.vue";
import routerList from "./routerList.vue";
import basePage from "./basePage.vue";
import logoTitle from "./logoTitle.vue";
import toGiteeIcoButton from "./toGiteeIcoButton.vue";
import toGithubIcoButton from "./toGithubIcoButton.vue";
import mobileGuideNavigationAids from "./mobileGuideNavigationAids.vue";
import titleAnchorNav from "./titleAnchorNav.vue";
import ActionLabel from "./ActionLabel.vue";
import DocCodeBox from "./DocCodeBox.vue";
import navigationLinks from "./navigationLinks.vue";

// import moduleName from './';

export function setCookie(name: string, value: any) {
  const Days = 30;
  const exp = new Date();
  exp.setTime(exp.getTime() + Days * 24 * 60 * 60 * 1000);
  document.cookie =
    name + "=" + encodeURI(value) + ";expires=" + exp.toUTCString();
}

// export function getCookie(name: string) {
//   let { arr, reg } = new RegExp("(^| )" + name + "=([^;]*)(;|$)"); //正则匹配
//   if ((arr = document.cookie.match(reg))) {
//     return decodeURI(arr[2]);
//   } else {
//     return null;
//   }
// }

// export function redColorModeCookie() {
//   /* 从Cookie 加载颜色主题设置,如为'null'则设置为 'true'即'Light Mode' */
//   const colorMode = Boolean(getCookie("colorMode"));
//   const autoColor = Boolean(getCookie("autoColor"));
//   if (colorMode === null) {
//     setColorModeCookie(true, false);
//     return true;
//   }
//   return {
//     darkMode: colorMode,
//     darkModeAuto: autoColor,
//   };
// }

// export function setColorModeCookie(color: Boolean, auto: Boolean) {
//   setCookie("colorMode", color ? "1" : "");
//   setCookie("autoColor", auto ? "1" : "");
// }

export {
  previewDemoCode,
  routerList,
  basePage,
  linkButton,
  logoTitle,
  titleAnchorNav,
  toGiteeIcoButton,
  toGithubIcoButton,
  mobileGuideNavigationAids,
  ActionLabel,
  DocCodeBox,
  navigationLinks,
};

import type { App } from "vue";
interface components {
  [key: string]: any;
}

export const components: components = {
  previewDemoCode,
  routerList,
  basePage,
  linkButton,
  logoTitle,
  titleAnchorNav,
  toGiteeIcoButton,
  toGithubIcoButton,
  mobileGuideNavigationAids,
  ActionLabel,
  DocCodeBox,
  navigationLinks,
};

export default (app: App) => {
  for (const key in components) {
    app.component(key, components[key]);
  }
};
