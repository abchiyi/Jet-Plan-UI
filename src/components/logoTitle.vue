<script lang="ts" setup>
import { linkButton } from "src/components";
</script>
<template>
  <link-button mode="text" to="/" aria-label="前往主页">
    <img src="/logo-mini.svg" alt="logo" />
    JetPlanUI
  </link-button>
</template>
<style scoped>
/* 标题按钮 */
.link-button {
  font-size: 1rem;
}
.link-button img {
  height: 1.5rem;
}
.link-button * {
  vertical-align: -0.35rem;
  display: inline-block;
}
</style>
