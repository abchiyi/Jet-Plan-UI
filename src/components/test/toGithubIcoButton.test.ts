import { describe, expect, test } from "vitest";
import { mount } from "@vue/test-utils";
import { Button } from "jet-plan-ui";
import { toGithubIcoButton } from "src/components";

const wrapper = mount(toGithubIcoButton);
describe("toGithubIcoButton", () => {
  test("MatchSnapshot", () => {
    expect(wrapper.html()).matchSnapshot();
  });
  test("Dom", () => {
    expect(wrapper.findComponent(Button).exists()).toBeTruthy();
    expect(wrapper.find("a").exists()).toBeTruthy();
  });
});
