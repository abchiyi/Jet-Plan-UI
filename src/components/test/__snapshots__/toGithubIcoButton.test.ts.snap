// Vitest Snapshot v1

exports[`toGithubIcoButton > MatchSnapshot 1`] = `
"<a class=\\"j-base-action j-action-feedback shape j-button text-button\\" ontouchstart=\\"function () { [native code] }\\" ontouchcancel=\\"function () { [native code] }\\" ontouchend=\\"function () { [native code] }\\" href=\\"https://github.com/abchiyi/Jet-Plan-UI\\" rel=\\"noopener noreferrer\\" aria-label=\\"前往github\\" target=\\"_blank\\"><i class=\\"bi bi-github\\"></i>
  <transition-stub name=\\"mask\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
    <!---->
  </transition-stub>
</a>"
`;
